### PR TITLE
[REVIEW] Create nyc_taxi_spatial_analysis notebook

### PR DESCRIPTION
**Pull request by taureandyernv**
This notebook demonstrates how to use cuSpatial's point in polygon to and shows its speedup by 
1. clean up pick up and drop off over 25 million points by finding points outside of the NYC boroughs land boundaries based on a shapefile of boroughs
1. compares it to a few points on CPU
1. uses that data to show which taxi rides were intra-borough and which were inter-borough 
1. visualizes it using cuXfilter

This is WIP, ready to start initial REVIEWs.  Please help refine while reviewing, as I'm sure iterative improvements can be made before merging @thomcom @exactlyallan @AjayThorve

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by taureandyernv** [comment]
@harrism @thomcom @AjayThorve @exactlyallan ready for review!

**Comment by harrism** [comment]
Our top priority for 0.14 is to get the libcudf++ port done. Appreciate your patience.

File: conda/environments/cuspatial_dev_cuda10.0.yml [hunk 1]
```diff
@@ -9,3 +9,4 @@ dependencies:
   - cudf=0.14.*
   - cudatoolkit=10.0
   - gdal>=2.4.*
+  - geopandas=0.7.0
```
Comment (raydouglass) [inline conda/environments/cuspatial_dev_cuda10.0.yml:12]: Is this dependency just for the notebook?

Also, it needs to be added to every environment file.

Comment (taureandyernv) [reply conda/environments/cuspatial_dev_cuda10.0.yml:12]: I'll add it to each env.  I think it is for this and any other notebook that uses geopandas.  This was an @thomcom add, so i trust its necessity, https://github.com/taureandyernv/cuspatial/commit/4e2c3ecfb1f9b83152f4acecf8200ba3fb218b2e#diff-5a99ec3304f1f4197b04cffc0fa980c1.

**Comment by taureandyernv** [comment]
@harrism I understand and good luck!  Just making known a hope :)

**Comment by taureandyernv** [comment]
Taurean's todo: 

- [ ] README edits, after code review and approval

**Comment by thomcom** [comment]
rerun tests

File: conda/environments/cuspatial_dev_cuda10.0.yml (hunk 1 above)
Comment (taureandyernv) [reply conda/environments/cuspatial_dev_cuda10.0.yml:14]: @raydouglass i think this was resolved.  can you check the changes?

Comment (thomcom) [reply conda/environments/cuspatial_dev_cuda10.0.yml:14]: Thanks for looking into this. I'm planning on making `geopandas` a regular dependency for the python portion of cuSpatial, and it made building this notebook easier.

**Comment by harrism** [comment]
Pushing to 0.15.

**Comment by harrism** [comment]
Pushing to 0.16 @taureandyernv

**Comment by cwharris** [comment]
@taureandyernv Did this notebook get included in the notebooks repo? Where does this PR stand?

**Comment by harrism** [comment]
I thought we were moving to include notebooks in their respective repos rather than have a central notebooks repo?

**Review by thomcom — APPROVED** [review]

**Pull request closed without merging**